cmd/gcp-controller-manager/node_csr_approver_test.go

```diff
@@ -1640,3 +1644,225 @@ func TestParseInstanceGroupURL(t *testing.T) {
+					},
+				},
+			},
+			expectedPod: &v1.Pod{
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: "default",
+					Name:      "testPod",
+				},
+				Spec: v1.PodSpec{
+					NodeName: "testNode",
+				},
+				Status: v1.PodStatus{
+					Phase: v1.PodFailed,
+					Conditions: []v1.PodCondition{
+						{
+							Type:    v1.DisruptionTarget,
+							Status:  v1.ConditionTrue,
+							Reason:  "DeletionByGCPControllerManager",
+							Message: "GCPControllerManager: node no longer exists",
+						},
+						{
+							Type:   v1.PodReady,
+							Status: v1.ConditionTrue,
+						},
+					},
+				},
+			},
+		},
+		{
+			desc: "Pod that was in Succeded phase should remain in Succeeded phase prior to deletion",
+			node: testNode,
+			pod: &v1.Pod{
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: "default",
+					Name:      "testPod",
+				},
+				Spec: v1.PodSpec{
+					NodeName: "testNode",
+				},
+				Status: v1.PodStatus{
+					Phase: v1.PodSucceeded,
+					Conditions: []v1.PodCondition{
+						{
+							Type:   v1.PodReady,
+							Status: v1.ConditionTrue,
+						},
+					},
+				},
+			},
+			expectedPod: &v1.Pod{
+				ObjectMeta: metav1.ObjectMeta{
+					Namespace: "default",
+					Name:      "testPod",
+				},
+				Spec: v1.PodSpec{
+					NodeName: "testNode",
+				},
+				Status: v1.PodStatus{
+					Phase: v1.PodSucceeded,
+					Conditions: []v1.PodCondition{
+						{
+							Type:    v1.DisruptionTarget,
+							Status:  v1.ConditionTrue,
+							Reason:  "DeletionByGCPControllerManager",
+							Message: "GCPControllerManager: node no longer exists",
+						},
+						{
+							Type:   v1.PodReady,
+							Status: v1.ConditionTrue,
+						},
+					},
+				},
+			},
+		},
+	} {
+		t.Run(tc.desc, func(t *testing.T) {
+
+			nodeList := &v1.NodeList{}
+			nodeList.Items = append(nodeList.Items, *tc.node)
+			podList := &v1.PodList{}
+			podList.Items = append(podList.Items, *tc.pod)
+
+			client := fake.NewSimpleClientset(nodeList, podList)
+			controllerCtx := &controllerContext{client: client}
+
+			err := deleteAllPodsBoundToNode(controllerCtx, tc.node.Name)
+
+			if err != nil {
+				t.Fatalf("Unexpected error deleting all pods bound to node %v", err)
+			}
+
+			actions := client.Actions()
+			if len(actions) != 3 {
+				t.Fatalf("Unexpected number of actions, got %v, want 3 (list pods, patch pod, delete pod)", len(actions))
+			}
+
+			patchAction, ok := actions[1].(testclient.PatchAction)
+			if !ok {
+				t.Fatalf("Unexpected object, got: %#v want: PatchAction", actions[1])
+			}
+
+			patchedPodBytes := patchAction.GetPatch()
+
+			originalPod, err := json.Marshal(tc.pod)
+			if err != nil {
+				t.Fatalf("Failed to marshal original pod %#v: %v", originalPod, err)
+			}
+			updated, err := strategicpatch.StrategicMergePatch(originalPod, patchedPodBytes, v1.Pod{})
+			if err != nil {
+				t.Fatalf("Failed to apply strategic merge patch %q on pod %#v: %v", patchedPodBytes, originalPod, err)
+			}
+			updatedPod := &v1.Pod{}
```

Comment:
Hmm, I don't think this would work because the pod is ultimately deleted, so I don't think it will be returned with a simple "GET" / "LIST" from the fakeClient. (unless there is a way to get deleted items from the fakeClient, but I'm not sure if that's supported?)

Reply:
I see, I suggest to keep as is then.